kii_thing_if.c

```diff
@@ -1395,6 +1410,64 @@ kii_bool_t get_thing_type(
     }
 }
 
+kii_bool_t update_thing_type(
+        kii_thing_if_t* kii_thing_if,
+        const char* thing_type,
+        kii_thing_if_error_t* error)
+{
+    switch (kii_thing_if->state) {
+        case KII_THING_IF_STATE_INITIALIZED:
+            if (error != NULL) {
+                error->reason = KII_THING_IF_ERROR_REASON_NOT_ONBOARDED;
+            }
+            return KII_FALSE;
+        case KII_THING_IF_STATE_STARTED:
+            if (error != NULL) {
+                error->reason = KII_THING_IF_ERROR_REASON_ALREADY_STARTED;
+            }
+            return KII_FALSE;
+        case KII_THING_IF_STATE_ONBOARDED:
+        {
+            char resource_path[128];
+            kii_t* kii = &(kii_thing_if->command_handler);
+
+            if (prv_set_thing_type_resource_path(
+                    kii,
+                    resource_path,
+                    sizeof(resource_path) / sizeof(resource_path[0])) != TRUE) {
+                return KII_FALSE;
+            }
+            if (prv_kii_api_call_start(
+                    kii,
+                    "PUT",
+                    resource_path,
+                    CONTENT_UPDATE_THING_TYPE,
+                    KII_TRUE,
+                    error) != 0) {
+                return KII_FALSE;
+            }
+            if (APPEND_BODY_CONST(kii, "{") != 0 ||
```

Comment:
@satoshikumano kumano san and I had oral discussion about this. As as result, we determined not to change this now.